docs/topics/multiplatform-mobile/multiplatform-mobile-upgrade-app.md

```diff
@@ -0,0 +1,348 @@
+</microformat>
+
+You've already implemented common logic using external dependencies. Now you can add more complex logic. Network 
+requests and data serialization are the [most popular cases](https://kotlinlang.org/lp/mobile/) to share with Kotlin
+Multiplatform. Learn how to implement them in your first application so that after completing this onboarding journey,
+you can use them in future projects.
+
+The updated app will retrieve data over the internet from a [SpaceX public API](https://docs.spacexdata.com/?version=latest)
+and display the date of the first successful launch of a SpaceX rocket.
+
+## Add more dependencies
+
+You'll need the following multiplatform libraries in your project:
+
+* [`kotlinx.coroutines`](https://github.com/Kotlin/kotlinx.coroutines) for using coroutines to write asynchronous code, thus allowing simultaneous operations
+* [`kotlinx.serialization`](https://github.com/Kotlin/kotlinx.serialization) for deserializing JSON responses into objects of entity classes used to process
+network operations
+* [Ktor](https://ktor.io/) framework as an HTTP client for retrieving data over the internet
+
+### kotlinx.coroutines
+
+To add `kotlinx.coroutines` to your project, specify a dependency in the common source set. For that, add the following
+line to the `build.gradle.kts` file of the shared module:
+
+```kotlin
+sourceSets {
+    commonMain {
+        dependencies {
+            // ...
+           implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.6.2")
+        }
+    }
+}
+```
+
+Multiplatform Gradle plugin automatically adds a dependency to the platform-specific (iOS and Android) parts
+of `kotlinx.coroutines`.
+
+You'll also use the new memory manager for Kotlin/Native, which will become default soon.
+Enable it in the same `build.gradle.kts` file:
+
+```kotlin
+val commonMain by getting {
+    // ...
+}
+
+kotlin.targets.withType(org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTarget::class.java) {
+    binaries.all {
+        binaryOptions["memoryModel"] = "experimental"
+    }
+}
+```
+
+### kotlinx.serialization
+
+For `kotlinx.serilization`, you need the plugin required by the build system. The Kotlin serialization plugin is shipped
+with the Kotlin compiler distribution, and the IntelliJ IDEA plugin is bundled into the Kotlin plugin.
+
+You can set up the serialization plugin with the Kotlin plugin using the Gradle plugins DSL by adding this line to
+the `plugins` block at the very beginning of the `build.gradle` file in the shared module:
+
+```kotlin
+plugins {
+    // 
+    kotlin("plugin.serialization") version "1.6.21"
+}
+```
+
+### Ktor
+
+You can add Ktor the same way you've added the `kotlinx.coroutines` library. In addition to specifying the core
```

Comment:
Let’s reword it with explicit references to dependencies because the current text forces me to think what the author means. Also, let’s don't use the word 'plugin' here. In the previous section, we used 'plugin' in a sense 'Gradle plugin' (at least, most of the time...), and here we use it in a completely different sense (it's Ktor plugin), and that's really confusing – we only add regular dependencies.
Smth like:

In addition to specifying the core `ktor-client-core` dependency in the common source set, you also need to:
* Add the ContentNegotiation functionality (`ktor-client-content-negotiation`) which is responsible for serializing/deserializing the content in a specific format. 
* Add the `ktor-serialization-kotlinx-json` depencency to instruct Ktor to use kotlinx.serialization as a serialisation library and the JSON format: to expect JSON data and deserialize it into a data class when receiving responses.
* Provide the platform engines by adding dependencies on the corresponding artifacts (`ktor-client-android`, `ktor-client-darwin`) in the platform source sets.